airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java

```diff
@@ -102,4 +91,11 @@ default ColumnId buildColumnId(final String name) {
    */
   String migrateFromV1toV2(StreamId streamId, String namespace, String tableName);
 
+  /**
+   * Typically we need to create a soft reset temporary table and clear loaded at values
+   *
+   * @return
+   */
+  String prepareTablesForSoftReset(final StreamConfig stream);
```

Comment:
nit: I'd mildly prefer to have the caller do a createTable call instead of making that implicit in this method (or at least have this method explicitly accept the table name suffix)